hooks/useAuth.ts

```diff
@@ -15,7 +16,9 @@ export const useAuth = () => {
     staleTime: Infinity,
   });
   const login = async (privkey: string) => {
-    const seckey = privkey.startsWith("nsec") ? decodeNsec(privkey) : privkey;
+    const seckey = privkey.startsWith("nsec")
+      ? decodeNsec(privkey)
+      : decodeNsec(encodeNsec(privkey) ?? ""); // encode and then decode hex privkey to make sure it is valid
```

Comment:
Should `""` be `null` instead? I think the conditional below would still pass with `""`

Reply:
decodeNsec takes a string and will return null if you pass an empty string cause that's not a valid nsec.

Reply:
also, the conditional would still fail if `seckey` was an empty string since an empty string is falsy.

Reply:
i mean the conditional would pass which is expected if seckey was an empty string. it should only make it pass the conditional if user enters a valid raw hex private key or a private key encoded as an nsec.